core/templates/dev/head/pages/exploration_editor/RouterServices.js

```diff
@@ -85,6 +85,17 @@ oppia.factory('routerService', [
       }
     });
 
+    // Select2 dropdowns does not automatically get populated
+    // when the tab is loaded, so we have to force it to refresh.
+    // Similar issue happened in Preferences.js and the same bug
+    // fix was made.
+    var _forceSelect2Refresh = function() {
+      $rootScope.select2DropdownIsShown = false;
```

Comment:
In general, try to never declare anything on $rootScope. Otherwise it just becomes a place to randomly stuff all sorts of things. Better to define this somewhere much more local, like the settings tab itself.